src/googmodule.ts

```diff
@@ -0,0 +1,388 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as path from 'path';
+
+import {ModulesManifest} from './modules_manifest';
+import {visitNodeWithSynthesizedComments} from './transformer_util';
+import * as ts from './typescript';
+
+export interface GoogModuleProcessorHost {
+  /**
+   * Takes a context (the current file) and the path of the file to import
+   *  and generates a googmodule module name
+   */
+  pathToModuleName(context: string, importPath: string): string;
+  /**
+   * If we do googmodule processing, we polyfill module.id, since that's
+   * part of ES6 modules.  This function determines what the module.id will be
+   * for each file.
+   */
+  fileNameToModuleId(fileName: string): string;
+  /** Whether to convert CommonJS module syntax to `goog.module` Closure imports. */
+  googmodule?: boolean;
+  /** Whether the emit targets ES5 or ES6+. */
+  es5Mode?: boolean;
+  /** expand "import 'foo';" to "import 'foo/index';" if it points to an index file. */
+  convertIndexImportShorthand?: boolean;
+
+  options: ts.CompilerOptions;
+  host: ts.ModuleResolutionHost;
+}
```

Comment:
I realize now that this is just a moved file, hmm.  Maybe it's not worth re-reviewing old code?  On the other hand maybe it's worth thinking about critically now that it's fresh?

Reply:
Yes, we should re-think what's needed here.